### ISSUE TITLE
Persona 3 Reload (2161700)

### ISSUE THREAD

**Issue by ciferkey** [ValveSoftware/Proton#7472]
# Compatibility Report
- Name of the game with compatibility issues: Persona 3 Reloaded
- Steam AppID of the game: 2161700 

## System Information
- GPU: AMD 7900 XT
- Video driver version: 23.1.9
- Kernel version: 6.7.2
- Link to full system information report as [Gist](https://gist.github.com/ciferkey/8581dcba9cf843df80d7e94cbaae0f22):
- Proton version: 8.0-5 and Experimental

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

[steam-2161700.log](https://github.com/ValveSoftware/Proton/files/14140720/steam-2161700.log)

## Symptoms 
Game immediately crashes to the crash report screen.

## Reproduction
No settings were changed from default settings. Launch the game directly from Steam.

**Comment by aarron-lee** [comment]
Just to add something I encountered. I'm running [Bazzite Deck Edition](https://github.com/ublue-os/bazzite/)

I have a GPD Win 4 6800u device on Bazzite, and it runs Persona 3 Reload in gamescope-session without any issues. No changes to args, etc. Just clicked play, and it worked in game mode.

I also have a Lenovo Legion Go with Bazzite, Z1 Extreme. Persona 3 Reload crashes immediately in gamescope-session, but works perfectly fine in Desktop mode (which is KDE on Wayland).

**Comment by ciferkey** [comment]
Reading through the bug thread on steam I saw this [comment](https://steamcommunity.com/app/2161700/discussions/0/4206993388793096605/#c4206993388793275861) which mentioned disabling additional monitors. If I start the game with only one monitor enabled it does not crash.

**Comment by waspennator** [comment]
Don't exactly have the game to check, but even with recent steamos updates and such, seeing people say that the game is still running like ass even with raytracing off in certain locations like Tartarus and the Velvet room.

**Comment by knosuhor** [comment]
Some light sources cause significant framedrops. Issue recorded on steam deck with recent os update, low settings, reflections off and VKD3D_CONFIG=nodxr just in case. Not an issue on windows.

https://github.com/ValveSoftware/Proton/assets/15619804/9e93ed98-35e9-4fe6-b182-115f859413bb

**Comment by vreh1** [comment]
@knosuhor I can confirm what he is saying, steamOS drops to insanely low fps with reflections on and off. I have friends that ran this game using windows and saw 0 issues, having stable 60 fps in these demanding areas. There are videos online of people using windows having much better framrate, with about a 40fps gain from not using steamOS. Please look into this. steamOS seems to be very buggy with persona 3 reload where windows is not.

---

Persona 3 Reload - Bad Framerate with steamOS opposed to Windows

Issue transferred from https://github.com/ValveSoftware/Proton/issues/7476.
@vreh1 posted on 2024-02-04T03:57:08:

Name of the game with compatibility issues: Persona 3 Reloaded
Steam AppID of the game: 2161700


System Information
GPU: AMD 7900 XT
Video driver version: 23.1.9
Kernel version: 6.7.2
Link to full system information report as [Gist](https://gist.github.com/ciferkey/8581dcba9cf843df80d7e94cbaae0f22):
Proton version: 8.0-5 and Experimental


I confirm:
 that I haven't found an existing compatibility report for this game.
 that I have checked whether there are updates for my system available.
[steam-2161700.log](https://github.com/ValveSoftware/Proton/files/14140720/steam-2161700.log)

Symptoms
Games framerate runs terribly using steamOS or latest version of proton. Using windows seems to fix all issues present in steamOS. 

Reproduction
No settings were changed from default settings. Launch the game directly from steamOS. dips down to 20 in places like tarturus and velvet room even below that. When using windows, these drops arent present and even perform well with reflections on

**Comment by Lacroixs** [comment]
@vreh1 I am hoping this gets fixed. Do we know how fast proton is for updates?

**Comment by vreh1** [comment]
@Lacroix I don't know really, I've only got into making github posts because of persona 3 reloads issues. From what I know its being looked at, I'd say wait 7 days and check back in

**Comment by Lacroixs** [comment]
Here, Game runs great on W11 on steam deck but chugs on proton. https://youtu.be/HQA_H4Kdbho?si=RyeDtsfJZ57TLFHn‬

**Comment by ciferkey** [comment]
Just to check when people say they are using a recent version of SteamOS do they specifically mean the  [3.5.14 preview](https://store.steampowered.com/news/app/1675200/view/4064004264594106055)?

The preview mentions:

> Improved performance of ray-traced reflections in Persona 3 Reload

**Comment by foxhull** [comment]
> Just to check when people say they are using a recent version of SteamOS do they specifically mean the [3.5.14 preview](https://store.steampowered.com/news/app/1675200/view/4064004264594106055)?
> 
> The preview mentions:
> 
> > Improved performance of ray-traced reflections in Persona 3 Reload

Yes I'm on the newest preview build. Disabling RT reflections does help in some areas, forcing Mesa 24 helps overall a small amount (~7 extra fps in problem spots of tartarus, nothing in velvet room). The big hitters now specifically are the light sources in the Velvet Room that come by every few seconds, teleporters in tartarus, and certain light sources in tartarus (windows, some chests, etc). So my theory at this point is that the reflections setting is working fine, but there's still some issue with the lighting that's not playing well with SteamOS sometimes. Otherwise the game runs great.

And to clarify, the lighting FPS issues are present on both Mesa 24 and SteamOS default, it's just that Mesa 24 helps the tartarus light sources not tank the frame rate quite as badly.

**Comment by knosuhor** [comment]
> Just to check when people say they are using a recent version of SteamOS do they specifically mean the [3.5.14 preview](https://store.steampowered.com/news/app/1675200/view/4064004264594106055)?
> 
> The preview mentions:
> 
> > Improved performance of ray-traced reflections in Persona 3 Reload

Yes, i am using 3.5.14 and proton experimental bleeding-edge. I don't think those framedrops are related to RT.

**Comment by Lacroixs** [comment]
The performance looks amazing all of sudden. Was there a hot fix?

**Comment by vreh1** [comment]
are you sure? can you check demanding areas?

**Comment by aarron-lee** [comment]
> I also have a Lenovo Legion Go with Bazzite, Z1 Extreme. Persona 3 Reload crashes immediately in gamescope-session, but works perfectly fine in Desktop mode (which is KDE on Wayland).

As a followup, I can get p3 reload to work in gamescope session via first booting up the game on an external monitor, and then disconnecting afterwards.

It's really weird how it can only boot up on an external screen first, but continues working on the internal screen afterwards.

edit: now fixed as of 07/13/2024

**Comment by aarron-lee** [comment]
Not sure when it's coming to proton, but looks like there's some upcoming work for a sizable performance increase in certain scenes on persona 3 reload: https://github.com/HansKristian-Work/dxil-spirv/pull/183

**Comment by Lacroixs** [comment]
> Not sure when it's coming to proton, but looks like there's some upcoming work for a sizable performance increase in certain scenes on persona 3 reload: [HansKristian-Work/dxil-spirv#183](https://github.com/HansKristian-Work/dxil-spirv/pull/183)

Who is that guy? Thank god though. Hopefully soon!

**Comment by Lacroixs** [comment]
@aarron-lee @vreh1 ITS FIXED... Go to Bleeding Edge beta of proton experimental, one of the devs fixed it. Was a shader compilation glitch.

**Comment by SiraMirai** [comment]
> @aarron-lee @vreh1 ITS FIXED... Go to Bleeding Edge beta of proton experimental, one of the devs fixed it. Was a shader compilation glitch.

It's in proton hotfix for me now as well. Improves performance for both steam deck and my desktop running fedora.

**Comment by waspennator** [comment]
Apparently Monad might still need some optimization checking around

**Comment by SMNFXCN** [comment]
Can confirm Monad has some performance dips.

I don't have footage but another issue I'm encountering is that P3RE specifically (haven't seen it on other games I have) breaks the ability to access Steam/QAM menus, won't let me adjust volume or activate sleep mode with the power button.
I've almost lost progress because the low battery toast didn't show up, I saved before that happened. Any toasts don't show up, achievements/screenshot etc.

I think I'm on Proton Experimental Bleeding Edge, steamOS Preview Channel/Beta Client
Edit: Proton Hotfix gives the same issues.

**Comment by waspennator** [comment]
Does this game have any more performance bottlenecks that proton or vkd3d can optimize out? Seems like its still having moments where it dips checking around

**Comment by SMNFXCN** [comment]
Anyone else having issues returning to Tartarus or going up floors? Sometimes I just get stuck on a black screen. Twice I've lost progress. Proton Experimental.
Also, I do not experience the steam overlay issues anymore
Edit: Clearing Proton files under BPM fixed this issue.
Since the steam overlay issues, I've been running Bazzite on my deck.
Update: I still get these random freezes. Sometimes when I start the game and load my save files.

Update 07/25/24: forgot to reply months ago, this has been resolved for me

**Comment by zany130** [comment]
![image](https://github.com/user-attachments/assets/aa324817-90b8-4360-9e9b-a879d2a0c57a)
getting graphic corruption when using gamescope's fsr

`RADV_DEBUG=nodcc`

fixes it 

should this be reported to gamescope?

Edit:sys info https://linux-hardware.org/?probe=1fdc847914

**Comment by space2ninja101** [comment]
My game crashes instantly 
heres my proton log
[steam-2161700.log](https://github.com/user-attachments/files/18271435/steam-2161700.log)

**Comment by kisak-valve** [comment]
Persona 3 Reload (2161700)

Issue transferred from https://github.com/ValveSoftware/Proton/issues/8356.
@space2ninja101 posted on 2024-12-30T01:10:24:

# Compatibility Report
- Name of the game with compatibility issues: Persona 3 Reload
- Steam AppID of the game: 2161700

## System Information
- GPU: GTX 1050 Ti
- Video driver version: nvidia 565.77
- Kernel version: 6.12.4-artix1-1
- Link to full system information report as https://gist.github.com/space2ninja101/52c4a70dc9f913ea4cf45e1588143459
- Proton version: Proton 9.0.4

Launch arguments: `PROTON_LOG=1 WINEDEBUG="+timestamp,+pid,+tid,+seh,+debugstr,+module" %command%`

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.


[steam-2161700.log](https://github.com/user-attachments/files/18271417/steam-2161700.log)
## Symptoms 
Game crashes immediately parsing shaders, disabled pre-caching and has same result.
No window pops up, just UE4's crash reporter.
Running with PROTON_USE_WINED3D=1 does however make a black fullscreen window appear, however it crashes after a few seconds. I have no plans to use WINED3D.

## Reproduction
Run Persona 3 Reload with Proton 9.0.4.


![image](https://github.com/user-attachments/assets/b79206f5-9929-4cd0-a60d-a43872dfdf50)
Window that comes up:

**Comment by space2ninja101** [comment]
really hope this becomes like active or something

**Comment by kisak-valve** [comment]
Hello @space2ninja101, I have a hypothesis that VKD3D-Proton is crashing while enumerating you Intel GPU. As a test, can you temporarily disable 64 bit mesa/ANV with the `VK_DRIVER_FILES` environment variable or temporarily rename the 64 bit mesa/ANV icd file in `/usr/share/vulkan/icd.d` to something that doesn't end in .json (like `intel_icd.x86_64.json.disabled`).

There's a potential fix at https://github.com/HansKristian-Work/vkd3d-proton/pull/2262.

**Comment by space2ninja101** [comment]
> Hello @space2ninja101, I have a hypothesis that VKD3D-Proton is crashing while enumerating you Intel GPU. As a test, can you temporarily disable 64 bit mesa/ANV with the `VK_DRIVER_FILES` environment variable or temporarily rename the 64 bit mesa/ANV icd file in `/usr/share/vulkan/icd.d` to something that doesn't end in .json (like `intel_icd.x86_64.json.disabled`).
> 
> There's a potential fix at [HansKristian-Work/vkd3d-proton#2262](https://github.com/HansKristian-Work/vkd3d-proton/pull/2262).

alright, but i have a suspicion its failing to parse shader cache, ill try though.

**Comment by space2ninja101** [comment]
> Hello @space2ninja101, I have a hypothesis that VKD3D-Proton is crashing while enumerating you Intel GPU. As a test, can you temporarily disable 64 bit mesa/ANV with the `VK_DRIVER_FILES` environment variable or temporarily rename the 64 bit mesa/ANV icd file in `/usr/share/vulkan/icd.d` to something that doesn't end in .json (like `intel_icd.x86_64.json.disabled`).
> 
> There's a potential fix at [HansKristian-Work/vkd3d-proton#2262](https://github.com/HansKristian-Work/vkd3d-proton/pull/2262).

HOLY FUCKING SHIT THANK YOU
